src/main/java/io/github/thebusybiscuit/slimefun4/implementation/tasks/armor/RainbowArmorTask.java

```diff
@@ -0,0 +1,56 @@
+package io.github.thebusybiscuit.slimefun4.implementation.tasks.armor;
+
+import javax.annotation.ParametersAreNonnullByDefault;
+
+import org.bukkit.Color;
+import org.bukkit.entity.Player;
+import org.bukkit.inventory.ItemStack;
+import org.bukkit.inventory.meta.LeatherArmorMeta;
+
+import io.github.thebusybiscuit.slimefun4.api.items.HashedArmorpiece;
+import io.github.thebusybiscuit.slimefun4.api.player.PlayerProfile;
+import io.github.thebusybiscuit.slimefun4.implementation.items.armor.RainbowArmorPiece;
+
+/**
+ * The {@link RainbowArmorTask} is responsible for handling the change in color of any Rainbow Armor piece.
+ *
+ * @author martinbrom
+ */
+public class RainbowArmorTask extends AbstractArmorTask {
+
+    private long currentColorIndex = 0;
+
+    @Override
+    protected void onTick() {
+        currentColorIndex++;
+    }
+
+    @Override
+    @ParametersAreNonnullByDefault
+    protected void onPlayerTick(Player p, PlayerProfile profile) {
+        for (int i = 0; i < 4; i++) {
+            ItemStack item = p.getInventory().getArmorContents()[i];
+
+            if (item != null && item.hasItemMeta()) {
+                HashedArmorpiece armorPiece = profile.getArmor()[i];
+
+                armorPiece.getItem().ifPresent(sfArmorPiece -> {
+                    if (sfArmorPiece instanceof RainbowArmorPiece && sfArmorPiece.canUse(p, true)) {
+                        updateRainbowArmor(item, (RainbowArmorPiece) sfArmorPiece);
+                    }
```

Comment:
Can we use instanceof casting here instead of casting the `sfArmorPiece` again when calling the method?